### PR TITLE
boards/olimex-msp430-h2618: add new board

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

This adds support of the Olimex MSP430-H2618 board. This board has plenty of RAM and ROM for an MSP430 board (even more than the z1), but still is easily obtainable.

### Testing procedure

1. Check the generated doc
2. Run a few test applications

### Issues/PRs references

- [x] depends on and includes https://github.com/RIOT-OS/RIOT/pull/19705

### CODE REVIEW

File: boards/olimex-msp430-h2618/include/board-conf.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+#ifndef BOARD_CONF_H
+#define BOARD_CONF_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define INFOMEM     (0x1000)
```
Comment (benpicco) [inline boards/olimex-msp430-h2618/include/board-conf.h:16]: I see all msp430 boards have this, but it's not actually used (and also present in the linkerscript) - what is this?

File: boards/olimex-msp430-h2618/include/board.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#ifndef __MSP430F2618__
+#define __MSP430F2618__
+#endif
+
+/**
+ * @name    Xtimer configuration
+ * @{
+ */
+#define XTIMER_WIDTH                (16)
+#define XTIMER_BACKOFF              (40)
```
Comment (benpicco) [inline boards/olimex-msp430-h2618/include/board.h:43]: Do we still need that?

File: boards/olimex-msp430-h2618/include/periph_conf.h [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * @ingroup     boards_olimex_msp430_h2618
+ * @{
+ *
+ * @file
+ * @brief       Olimex-MSP430-H2618 peripheral configuration
+ *
+ * @author      Oliver Hahm <oliver.hahm@inria.fr>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
```
Comment (benpicco) [inline boards/olimex-msp430-h2618/include/periph_conf.h:18]: I don't think those two have that board :wink:

File: boards/olimex-msp430-h2618/include/board.h (hunk 2 above)
Comment (maribu) [reply boards/olimex-msp430-h2618/include/board.h:43]: As long as we don't drop the xtimer implementation of the xtimer API, we still need this :/

**Review by benpicco — APPROVED** [review]

**Comment by maribu** [comment]
bors merge

**Comment by maribu** [comment]
bors retry

**Comment by maribu** [comment]
bors retry

**Comment by maribu** [comment]
bors retry

**Pull request merged**